File: Aula_0602/ex01.cs
using System;

class Pais {
  private string nome = "Nome do país";
  private int populacao = 1;
  // public Pais() { }
  public Pais(string nome, int populacao) {
    // SetNome(nome);
    // SetPopulacao(populacao);
    if (nome != "") this.nome = nome;
    if (populacao > 0) this.populacao = populacao;
  }
  public void SetNome(string nome) {
    if (nome != "") this.nome = nome;
  }
  public void SetPopulacao(int populacao) {
    if (populacao > 0) this.populacao = populacao;
  }
  public string GetNome() {
    return this.nome;
  }
  public int GetPopulacao() {
    return this.populacao;
  }
  public override string ToString() {
    return $"Nome: {nome} - {populacao} habitante(s)";
  }
}

class Program {
  public static void Main() {
    Console.WriteLine("Informe o nome do 1º país e sua população");
    string nome = Console.ReadLine();
    int pop = int.Parse(Console.ReadLine());
    Pais maispop = new Pais(nome, pop);
    Console.WriteLine(maispop);
    for (int i = 2; i <= 10; i++) {
      Console.WriteLine($"Informe o nome do {i}º país e sua população");
      nome = Console.ReadLine();
      pop = int.Parse(Console.ReadLine());
      Pais p = new Pais(nome, pop);
      Console.WriteLine(p);
      if (p.GetPopulacao() > maispop.GetPopulacao()) maispop = p;
    }
    Console.WriteLine($"País mais populoso: {maispop}");
  }
}

File: Aula_0602/ex01.py
class Pais:
  def __init__(self, nome, populacao):
    self.__nome = "Nome do país"
    self.__populacao = 1
    if nome != "": self.__nome = nome
    if populacao > 0: self.__populacao = populacao
  def set_nome(self, nome):
    if nome != "": self.__nome = nome
  def set_populacao(self, populacao):
    if populacao > 0: self.__populacao = populacao
  def get_nome(self):
    return self.__nome
  def get_populacao(self):
    return self.__populacao
  def __str__(self):
    return f"Nome: {self.__nome} - {self.__populacao} habitante(s)"

if __name__ == "__main__":
  print("Informe o nome do 1º país e sua população")
  nome = input()
  pop = int(input())
  maispop = Pais(nome, pop)
  print(maispop)
  for i in range(2, 4):
    print(f"Informe o nome do {i}º país e sua população")
    nome = input()
    pop = int(input())
    p = Pais(nome, pop)
    print(p)
    if p.get_populacao() > maispop.get_populacao(): maispop = p
  print(f"País mais populoso: {maispop}")



